oak_remote_attestation/proto/v1/evidence.proto

```diff
@@ -0,0 +1,119 @@
+// limitations under the License.
+//
+
+syntax = "proto3";
+
+package oak.remote_attestation.v1;
+
+option java_multiple_files = true;
+option java_package = "com.google.oak.remote_attestation.v1";
+
+// This proto defines the layered DICE Attestation Evidence.
+//
+// DICE provides a mechanism for combining software measurements and corresponding certificates into
+// a chain, where each element (called “layer”) represents a piece of software loaded into the VMs
+// memory. These layers are loaded sequentially, i.e. the previous layer loads the next layer. The
+// previous layer is also responsible for measuring the next layer, generating a private key and a
+// certificate for it.
+//
+// <https://trustedcomputinggroup.org/wp-content/uploads/TCG_DICE_Attestation_Architecture_r22_02dec2020.pdf>
+// <https://trustedcomputinggroup.org/wp-content/uploads/DICE-Layering-Architecture-r19_pub.pdf>
+
+enum TeePlatform {
+  UNSPECIFIED = 0;
+  AMD_SEV_SNP = 1;
+  INTEL_TDX = 2;
+}
+
+message RootLayerEvidence {
+  // The platform providing the attestation report.
+  TeePlatform platform = 0;
+  // TEE-specific attestation report acting as a non-standard certificate for the
+  // Layer0 ECA public key.
+  bytes remote_attestation_report = 1;
+  // Serialised ECA public key for Layer0.
+  bytes eca_public_key = 2;
+}
+
+message LayerEvidence {
+  string layer_name = 0;
+  // CBOR/COSE/CWT ECA certificate.
+  // <https://www.rfc-editor.org/rfc/rfc8392.html>
+  bytes eca_certificate = 1;
+}
+
+message ApplicationKeys {
+  // CBOR/COSE/CWT encryption public key certificate.
+  // <https://www.rfc-editor.org/rfc/rfc8392.html>
+  optional bytes encryption_public_key_certificate = 0;
+  // CBOR/COSE/CWT signing public key certificate.
+  // <https://www.rfc-editor.org/rfc/rfc8392.html>
+  optional bytes signing_public_key_certificate = 1;
+}
+
+message Evidence {
+  // Layer0 attestation evidence.
+  RootLayerEvidence root_layer_evidence = 0;
+  // Layer1..LayerN-1 attestation evidence.
+  repeated LayerEvidence layer_evidence = 1;
+  // Final layer keys.
+  optional ApplicationKeys application_keys = 2;
+}
+
+// Endorsement contains statements that some entity (e.g., a hardware provider) vouches for the
+// integrity of claims about the TEE or the software running on it.
+// The name is chosen to match the RATS terminology:
+// <https://www.rfc-editor.org/rfc/rfc9334.html#name-endorsements>
+message Endorsement {
+  // The serialized TEE certificate(s). The details of the format and how the certificate(s) are
+  // encoded into this byte array are implementation-specific. In case of AMD-SEV-SNP, as described
+  // in https://www.amd.com/system/files/TechDocs/57230.pdf, there are three different certificates
+  // packaged in two different files.
+  bytes tee_certificates = 1;
+
+  // Transparent Release endorsement.
+  TransparentReleaseEndorsement transparent_release_endorsement = 2;
+}
+
+message TransparentReleaseEndorsement {
```

Comment:
Based on my earlier point that there are likely to be multiple endorsements it might be useful to add an identifier to indicate which components this applies to.

Reply:
I agree. The endorsement statement contains the binary name. We can either use that for matching a layer name or add an explicit new field here. The latter might give more flexibility.